ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -987,10 +989,14 @@ public void createTable(Table tbl, boolean ifNotExists,
           tTbl.setPrivileges(principalPrivs);
         }
       }
-      // Set table snapshot to api.Table to make it persistent.
-      TableSnapshot tableSnapshot = AcidUtils.getTableSnapshot(conf, tbl, true);
-      if (tableSnapshot != null) {
-        tbl.getTTable().setWriteId(tableSnapshot.getWriteId());
+      // Set table snapshot to api.Table to make it persistent. A transactional table being
+      // replicated may have a valid write Id copied from the source. Use that instead of
+      // crafting one on the replica.
+      if (tTbl.getWriteId() <= 0) {
```

Comment:
DO_NOT_UPDATE_STATS flag should be set in createTableFlow as well. Or else in autogather mode at target, it will be updated automatically.

Reply:
You are right. We do not need it at the creation time. We already have tests for that and they are working fine i.e. the expected stats both the table level and column level is getting replicated.